### PR TITLE
Inference pass

### PR DESCRIPTION
**Pull request by szmazurek**
Fixes #ISSUE_NUMBER

## Proposed Changes

- 

## Checklist

- [ ] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [ ] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [ ] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [ ] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [ ] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].
- [ ] The `logging` library is being used and no `print` statements are left.

### CODE REVIEW

File: GANDLF/compute/forward_pass.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def validate_network(
                 label_ground_truth = torch.cat(
                     [subject[key] for key in params["value_keys"]], dim=0
                 )
+                # Here I think the code is missing the following line:
+                # label_present = True
```
Comment (sarthakpati) [inline GANDLF/compute/forward_pass.py]: Nope, this logic is complete.

- [These lines](https://github.com/szmazurek/GaNDLF/blob/02ec726f752a80cbb61de9d940d2adc3cf5bf21f/GANDLF/compute/forward_pass.py#L130-L138) check for the presense of the `"label"` header, upon which the `label_present` flag gets set. 
- [These lines](https://github.com/szmazurek/GaNDLF/blob/02ec726f752a80cbb61de9d940d2adc3cf5bf21f/GANDLF/compute/forward_pass.py#L140-L145) only check if the `"ValueToPredict"` header is present or not (I agree this is not elegant, and we need to figure out a better mechanism).
- [These lines](https://github.com/szmazurek/GaNDLF/blob/02ec726f752a80cbb61de9d940d2adc3cf5bf21f/GANDLF/compute/forward_pass.py#L156-L216) are for inference for classification/regression only when the `"label"` header is defined.
- [These lines](https://github.com/szmazurek/GaNDLF/blob/02ec726f752a80cbb61de9d940d2adc3cf5bf21f/GANDLF/compute/forward_pass.py#L218-L305) are for inference with classification/regression when the `"label"` header is not present, or for segmentation (in which case the `"label"` header will anyway not be present).

Does this help clarify?

Comment (szmazurek) [reply GANDLF/compute/forward_pass.py]: Yeah, thanks. Deleting the comment
